rules/naming/src/PropertyRenamer/AbstractPropertyRenamer.php

```diff
@@ -0,0 +1,140 @@
+use PhpParser\Node\Expr\StaticPropertyFetch;
+use PhpParser\Node\Identifier;
+use PhpParser\Node\Stmt\Property;
+use PhpParser\Node\VarLikeIdentifier;
+use Rector\Core\PhpParser\NodeTraverser\CallableNodeTraverser;
+use Rector\Naming\Guard\DateTimeAtNamingConventionGuard;
+use Rector\Naming\Guard\GuardInterface;
+use Rector\Naming\Guard\HasMagicGetSetGuard;
+use Rector\Naming\Guard\NotPrivatePropertyGuard;
+use Rector\Naming\Guard\RamseyUuidInterfaceGuard;
+use Rector\Naming\RenameGuard\PropertyRenameGuard;
+use Rector\Naming\RenameGuard\RenameGuardInterface;
+use Rector\Naming\ValueObject\PropertyRename;
+use Rector\NodeNameResolver\NodeNameResolver;
+
+abstract class AbstractPropertyRenamer
+{
+    /**
+     * @var RenameGuardInterface
+     */
+    protected $propertyRenameGuard;
+
+    /**
+     * @var GuardInterface
+     */
+    protected $conflictingPropertyNameGuard;
+
+    /**
+     * @var CallableNodeTraverser
+     */
+    private $callableNodeTraverser;
+
+    /**
+     * @var NodeNameResolver
+     */
+    private $nodeNameResolver;
+
+    /**
+     * @var NotPrivatePropertyGuard
+     */
+    private $notPrivatePropertyGuard;
+
+    /**
+     * @var RamseyUuidInterfaceGuard
+     */
+    private $ramseyUuidInterfaceGuard;
+
+    /**
+     * @var DateTimeAtNamingConventionGuard
+     */
+    private $dateTimeAtNamingConventionGuard;
+
+    /**
+     * @var HasMagicGetSetGuard
+     */
+    private $hasMagicGetSetGuard;
+
+    /**
+     * @required
+     */
+    public function autowireAbstractPropertyRenamer(
+        CallableNodeTraverser $callableNodeTraverser,
+        NodeNameResolver $nodeNameResolver,
+        NotPrivatePropertyGuard $notPrivatePropertyGuard,
+        RamseyUuidInterfaceGuard $ramseyUuidInterfaceGuard,
+        DateTimeAtNamingConventionGuard $dateTimeAtNamingConventionGuard,
+        PropertyRenameGuard $propertyRenameGuard,
+        HasMagicGetSetGuard $hasMagicGetSetGuard
+    ): void {
+        $this->callableNodeTraverser = $callableNodeTraverser;
+        $this->nodeNameResolver = $nodeNameResolver;
+        $this->notPrivatePropertyGuard = $notPrivatePropertyGuard;
+        $this->ramseyUuidInterfaceGuard = $ramseyUuidInterfaceGuard;
+        $this->dateTimeAtNamingConventionGuard = $dateTimeAtNamingConventionGuard;
+        $this->propertyRenameGuard = $propertyRenameGuard;
+        $this->hasMagicGetSetGuard = $hasMagicGetSetGuard;
+    }
+
+    public function rename(PropertyRename $propertyRename): ?Property
+    {
+        if ($this->areNamesDifferent($propertyRename)) {
+            return null;
+        }
+
+        if ($this->propertyRenameGuard->shouldSkip($propertyRename, [
+            $this->notPrivatePropertyGuard,
+            $this->conflictingPropertyNameGuard,
+            $this->ramseyUuidInterfaceGuard,
+            $this->dateTimeAtNamingConventionGuard,
+            $this->hasMagicGetSetGuard,
+        ])) {
+            return null;
+        }
```

Comment:
Don't you think it might be a good idea to decouple this logic to `shouldSkip` method in this class so that it could be possible to override the guards in particular renamers when needed? For example, for UnderscoreCamelCasePropertyRenamer we can add an additional guard which will skip renaming logic if there is no `_` in the name. 
Another possibility is to add the ability to pass additional guards directly from the rule itself.

Reply:
Yes, there is similar case for some type resolver. It allows to exclude one or more type resolvers based on type.

Reply:
By "Yes" you mean the first option with overriding or to pass additional guards directly as a method param? I bet second because I saw something like this somewhere.

Reply:
I prefer visual lists with numbers, if there are more options :D

2nď I guess, I'd have to see it